### PR TITLE
Resolves #42 #41 #40, also resolves a bug with virtualbox on windows

### LINKED ISSUE #42: firecracker crashes if dhcp not running
Scanning inputs  ⠋                                                                                
Writing image done [==============================================================================] 652.0 MiB / 652.0 MiB
Error: Run 'sudo vorteil firecracker-setup' for the listener
^Cpanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x188 pc=0x21f1b06]

goroutine 16 [running]:
github.com/firecracker-microvm/firecracker-go-sdk.(*Machine).Shutdown(0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jensg/go/pkg/mod/github.com/firecracker-microvm/firecracker-go-sdk@v0.21.0/machine.go:410 +0x26
github.com/vorteil/vorteil/pkg/virtualizers/firecracker.(*Virtualizer).Stop(0xc000956dc0, 0x297b368, 0x0)
	/home/jensg/go/src/github.com/vorteil/vorteil/pkg/virtualizers/firecracker/virtualizer_linux.go:528 +0xbc
main.run.func2(0x2d36900, 0xc000956dc0, 0xc0004a8770)
	/home/jensg/go/src/github.com/vorteil/vorteil/cmd/vorteil/run.go:257 +0x35
created by main.run
	/home/jensg/go/src/github.com/vorteil/vorteil/cmd/vorteil/run.go:256 +0x7bb

### PR DESCRIPTION
**Pull request by trentis**
#42 #41 #40

### CODE REVIEW

File: pkg/virtualizers/firecracker/virtualizer_linux.go [hunk 1]
```diff
@@ -88,6 +89,9 @@ func SetupBridgeAndDHCPServer() error {
 	go func() {
 		http.ListenAndServe(":7476", nil)
 	}()
+	tempLog := elog.CLI{}
```
Comment (jalfvort) [inline pkg/virtualizers/firecracker/virtualizer_linux.go:92]: Why create a logger, shouldn't fmt.Printf() be good enough?

**Review by jalfvort — COMMENTED** [review]
Tested and working on Linux.

File: pkg/virtualizers/firecracker/virtualizer_linux.go (hunk 1 above)
Comment (trentis) [reply pkg/virtualizers/firecracker/virtualizer_linux.go:92]: Decided to use our library over the standard. I guess printf is technically just better in general

**Comment by jalfvort** [comment]
There is one additional problem discovered that somewhat relates to issue #40 . It looks like the earlier than the kernel manager or whatever merges the default kernel values. Here is an example of what kind of problem this causes:
### Scenario 1 - Not Kernel defined :
**VCFG**
```toml
....
[vm]
  cpus = 1
  ram = "91 MiB"
  inodes = 100
  disk-size = "+10 MiB"
```

```sh
.$ /cli run /tmp/helloworld/ --platform firecracker
Scanning inputs  ⠋                                                                                
Writing image done [=================================] 32.0 MiB / 32.0 MiB
Kernel 'firecracker-' VMLinux does not exist
```
_visualizers logic thinks kernel is = "" , but it should be whatever the default / latest kernel is._

### Scenario 2 - Old Kernel defined :
**VCFG**
```toml
....
[vm]
  cpus = 1
  ram = "91 MiB"
  inodes = 100
  kernel = "20.8.3"
  disk-size = "+10 MiB"
```

```sh
$ ./cli run /tmp/helloworld/ --platform firecracker
Scanning inputs  ⠋                                                                                
Requested kernel '20.8.3' is too old for this compiler. Using latest kernel instead.
Writing image done [==============================] 32.0 MiB / 32.0 MiB
Kernel 'firecracker-20.8.3' VMLinux does not exist
```
_visualizers logic thinks kernel is = "20.8.3" (from vcfg) , but it should be whatever the default / latest kernel is (kernel manger is setting this)._

I can approve the pr for now if you want and make this another issue, but I thought I would see if you wanted to fix this issue in this pr.

**Pull request merged**